server/routes.py

```diff
@@ -45,11 +77,54 @@ def normalize(
     }
 
     try:
-        _normalize(
+        (
+            executor,
+            docstring,
+            init,
+            endpoints,
+            filepath,
+            hubble_score_metrics,
+        ) = _normalize(
             block_data.package_path,
             meta=block_data.meta,
             env=block_data.env,
         )
+        if init:
+            init_args, init_kwargs, init_docstring = init
+            init = {
+                'args': [
+                    {'arg': arg, 'annotation': annotation}
+                    for arg, annotation in init_args
+                ],
+                'kwargs': [
+                    {'arg': arg, 'annotation': annotation, 'default': default}
+                    for arg, annotation, default in init_kwargs
+                ],
+                'docstring': init_docstring,
+            }
+        result['data'] = {
+            'executor': executor,
+            'docstring': docstring,
+            'init': init,
+            'endpoints': [
+                {
+                    'name': endpoint_name,
+                    'args': [
+                        {'arg': arg, 'annotation': annotation}
+                        for arg, annotation in endpoint_args
+                    ],
+                    'kwargs': [
+                        {'arg': arg, 'annotation': annotation, 'default': default}
+                        for arg, annotation, default in endpoint_kwargs
+                    ],
+                    'docstring': endpoint_docstring,
+                    'requests': endpoint_requests,
+                }
```

Comment:
For this `arg` and `kwargs` setting - why not already return outputs formatted like this in `_normalize`? Since this is the only place the outputs from that function are used.

In fact it seems to me this whole code block  could (in try) be basically a pass-through, if you slightly restructure the return value of `normalize`

Reply:
refactored